analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,248 @@
+                    ArrayCreationExpressionSyntax arrayCreation => ArrayCreationPostProcess(arrayCreation, programState),
+                    AssignmentExpressionSyntax assignment => AssignmentExpressionPostProcess(assignment, programState),
+                    _ => programState
+                };
+
+                return base.PostProcessInstruction(programPoint, programState);
+            }
+
+            public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                if (programPoint.CurrentInstruction is InvocationExpressionSyntax invocation)
+                {
+                    programState = InvocationExpressionPreProcess(invocation, programState);
+                }
+
+                return base.PreProcessInstruction(programPoint, programState);
+            }
+
+            private ProgramState ObjectCreationPostProcess(ObjectCreationExpressionSyntax objectCreation, ProgramState programState)
+            {
+                if (semanticModel.GetSymbolInfo(objectCreation).Symbol is {} symbol
+                    && symbol.ContainingType is {} symbolType
+                    && symbolType.IsAny(VulnerableTypes))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.NotInitialized);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.NotInitialized);
+                }
+
+                return programState;
+            }
+
+            private ProgramState InvocationExpressionPostProcess(InvocationExpressionSyntax invocation, ProgramState programState)
+            {
+                if (IsCreateMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.NotInitialized);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.NotInitialized);
+                }
+                else if (IsGenerateKeyMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = GetSymbolicValue(invocation, programState);
+                    programState = programState.SetConstraint(symbolicValue, KeyInitializationSymbolicValueConstraint.Initialized);
+                }
+                else if (IsGenerateIVMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = GetSymbolicValue(invocation, programState);
+                    programState = programState.SetConstraint(symbolicValue, IVInitializationSymbolicValueConstraint.Initialized);
+                }
+                else if (IsRngCryptoServiceProviderSanitizer(invocation, semanticModel)
```

Comment:
There should be two sanitizer APIs, `RngCryptoServiceProvider` and `RandomNumberGenerator`, one of them uses the other under the hood.

Example usage is in visible in another RSPEC: https://github.com/SonarSource/security-expected-issues/blob/master/dotnet/NetCore3MVC/Controllers/RSPEC_2053/RSPEC2053_Compliant.cs

Reply:
Indeed. I've missed `RandomNumberGenerator` since it wasn't in rule specs.